maven-repository-metadata/src/main/mdo/metadata.mdo

```diff
@@ -202,6 +202,30 @@ under the License.
                         changed = true;
                     }
                 }
+
+                for ( SnapshotVersion snapshotVersion : versioning.getSnapshotVersions() )
+                {
+                    boolean exists = false;
+
+                    for ( SnapshotVersion sv : v.getSnapshotVersions() )
+                    {
+                        if ( java.util.Objects.equals( snapshotVersion.getClassifier(), sv.getClassifier() ) &&
+                             java.util.Objects.equals( snapshotVersion.getExtension(), sv.getExtension() ) &&
+                             java.util.Objects.equals( snapshotVersion.getUpdated(), sv.getUpdated() ) &&
```

Comment:
This is not correct, as only classifier, extension and version act as key. The updated may actually differ but the should not lead to duplicate keys.